src/wallet/db.cpp

```diff
@@ -463,7 +463,8 @@ BerkeleyBatch::BerkeleyBatch(BerkeleyDatabase& database, const char* pszMode, bo
         if (!env->Open(false /* retry */))
             throw std::runtime_error("BerkeleyBatch: Failed to open database environment.");
 
-        pdb = env->mapDb[strFilename];
+        BerkeleyDatabase& database = env->m_databases.at(strFilename).get();
```

Comment:
Some comments here:
 - there's a slight change when replacing `map::operator[]` with `map::at`, maybe worth a comemnt or an assertion?
 - also, isn't this equivalent to the above `database` argument?
 - is this is really necessary then you could avoid shadowing `database`.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/14552#discussion_r230800120

Oh, I didn't even realize there was a `database` argument being shadowed above. Should just delete this line and use the existing `database`, which points to the same thing.